pkg/virt-controller/watch/workload-updater/workload-updater.go

```diff
@@ -313,14 +321,27 @@ func (c *WorkloadUpdateController) doesRequireMigration(vmi *virtv1.VirtualMachi
 	if vmi.IsFinal() || migrationutils.IsMigrating(vmi) {
 		return false
 	}
-
+	if metav1.HasAnnotation(vmi.ObjectMeta, v1.WorkloadUpdateMigrationAbortionAnnotation) {
+		return false
+	}
 	if isHotplugInProgress(vmi) {
 		return true
 	}
 
 	return false
 }
 
+func (c *WorkloadUpdateController) shouldAbortMigration(vmi *virtv1.VirtualMachineInstance) bool {
+	numMig := len(migrationutils.ListWorkloadUpdateMigrations(c.migrationInformer.GetStore(), vmi.Name, vmi.Namespace))
```

Comment:
is it worth ensuring that we are only aborting the migration this controller started?
```_, ok = migration.Annotations[virtv1.WorkloadUpdateMigrationAnnotation]```

Reply:
It is already the case. The function `ListWorkloadUpdateMigrations` checks if the migration object has the annotation `WorkloadUpdateMigrationAnnotation` (see [here](https://github.com/kubevirt/kubevirt/blob/4dfde6f41a9a4603c8f8764d5ee0f8777e54ad87/pkg/util/migrations/migrations.go#L42)), otherwise the migration isn't included in the list.

Reply:
I find it a cleaner encapsulated in the function, and the function it is also used below

Reply:
I missed it! Thanks for pointing me to it!